cardano-node/src/Cardano/Node/Types.hs

```diff
@@ -355,6 +350,54 @@ data NodeHardForkProtocolConfiguration =
        -- configured the same, or they will disagree.
        --
      , npcTestShelleyHardForkAtVersion :: Maybe Word
+
+       -- | If we have knowledge about when the Allegra hard fork is then we
+       -- have an opportunity to optimise the bulk sync slightly.
+       --
+     , npcAllegraHardForkNotBeforeEpoch :: Maybe EpochNo
+
+       -- | For testing purposes we support specifying that the hard fork
+       -- happens at an exact epoch number (ie the first epoch of the new era).
+       --
+       -- Obviously if this is used, all the nodes in the test cluster must be
+       -- configured the same, or they will disagree.
+       --
+     , npcTestAllegraHardForkAtEpoch :: Maybe EpochNo
```

Comment:
We can perhaps simplify this by simply referring to the comment for the other field, rather than repeating everything many times.